vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go

```diff
@@ -334,39 +328,29 @@ func resourceVcdNsxtIpSecVpnTunnelUpdate(ctx context.Context, d *schema.Resource
 	// Inject ID for update
 	ipSecVpnConfig.ID = d.Id()
 
-	securityProfileHasChange := d.HasChange("security_profile_customization")
-	_, newSecurityProfile := d.GetChange("security_profile_customization")
+	_, isSet := d.GetOk("security_profile_customization")
 
-	// Security Profile Customization settings work on two different endpoints:
-	// * To set a custom security profile - there is a separate endpoint where all security profile settings can be
-	// set. After setting them, parent IPsec VPN Tunnel `SecurityType` becomes "CUSTOM".
-	// * To remove customization and switch back to NSX-T Default parameters the parent IPsec VPN Tunnel must be updated
-	// and its field 'SecurityType' must be set to 'DEFAULT'
-	if securityProfileHasChange && len(newSecurityProfile.([]interface{})) == 0 {
+	// If unset, it will set it back to default, so before updating the VPN config, we need to check if the `security_profile_customization`
+	// field is defined in the configuration
+	if isSet {
+		ipSecVpnConfig.SecurityType = "CUSTOM"
+	} else {
 		ipSecVpnConfig.SecurityType = "DEFAULT"
 	}
 
-	// At first update IPsec VPN tunnel configuration
-	// It will reset Security Profile to DEFAULT at the same shot if no customization exists in 'security_profile_customization'
 	updatedIpSecVpnConfiguration, err := existingIpSecVpnConfiguration.Update(ipSecVpnConfig)
 	if err != nil {
 		return diag.Errorf("[nsx-t ipsec vpn tunnel update] error updating NSX-T IPsec VPN Tunnel configuration '%s': %s", ipSecVpnConfig.Name, err)
 	}
 
-	// If Security Profile has change and it is being customized
-	if securityProfileHasChange && newSecurityProfile != nil {
```

Comment:
So if I understood, this condition is the culprit as `existingIpSecVpnConfiguration.Update(ipSecVpnConfig)` restarts the security profile config?

Reply:
Yes. Basically, if there was no change to the security profile, only to the vpn tunnel itself, it would update the tunnel with a blank `SecurityType` which resulted in it getting reset to `DEFAULT`

Reply:
Thanks for your answer!